File: src/my_py_pkg/my_py_pkg/hardware_status_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus

class HardwareStatusPublisherNode(Node):
  def __init__(self):
    super().__init__("hardware_status_publisher")
    self.hw_status_pub = self.create_publisher(HardwareStatus, "hw_sts", 10)
    self.create_timer(timer_period_sec=1.0, callback=self.timer_callback)
    self.get_logger().info("HW status publisher has been started")
    
  def timer_callback(self):
    msg = HardwareStatus()
    msg.temperature = 43.6
    msg.are_motors_ready = True
    msg.debug_message = "Nothing Special"
    self.hw_status_pub.publish(msg)

def main(args=None):
  rclpy.init(args=args)
  node = HardwareStatusPublisherNode()
  rclpy.spin(node)
  rclpy.shutdown()
  
if __name__ == "__main__":
  main()

File: src/my_cpp_pkg/src/hardware_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp"

class HardwareStatusPublisherNode : public rclcpp::Node
{
public:
  HardwareStatusPublisherNode():Node("hw_sts_pub")
  {
    this->puber_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("hw_sts", 10);
    this->timer_ = this->create_wall_timer(std::chrono::seconds(1), 
                                      std::bind(&HardwareStatusPublisherNode::timerCallback, this));
    RCLCPP_INFO(this->get_logger(), "Hardware Status Publisher has been started");
  }
private:      
  void timerCallback()
  {
    auto msg = my_robot_interfaces::msg::HardwareStatus();
    msg.temperature = 88.88;
    msg.are_motors_ready = true;
    msg.debug_message = "Nothing Special";
    this->puber_->publish(msg);
  }
  rclcpp::TimerBase::SharedPtr timer_;
  rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr puber_;

};


int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<HardwareStatusPublisherNode>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}